### PR TITLE
Cleanup dead ENABLE_WASM_TELEMETRY

### PR DESCRIPTION
**Pull request by howardjohn**
This doesn't actually work anyways, so no release note needed

### CODE REVIEW

File: pilot/pkg/model/wasm.go [hunk 1]
```diff
@@ -17,41 +17,20 @@ package model
...
 	wasm "github.com/envoyproxy/go-control-plane/envoy/extensions/wasm/v3"
-
-	"istio.io/istio/pilot/pkg/features"
 )
 
-// ConstructVMConfig constructs a VM config. If WASM is enabled, the wasm plugin at filename will be used.
-// If not, the builtin (null vm) extension, name, will be used.
-func ConstructVMConfig(filename, name string) *wasm.PluginConfig_VmConfig {
-	var vmConfig *wasm.PluginConfig_VmConfig
-	if features.EnableWasmTelemetry && filename != "" {
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/wasm.go:28]: missed this env?

**Review by jacob-delgado — APPROVED** [review]

File: pilot/pkg/model/wasm.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/model/wasm.go:28]: Not sure what you mean?

Comment (kyessenov) [reply pilot/pkg/model/wasm.go:28]: @hzxuzhonghu there's no need to have a special telemetry-wasm flag. Users should use Wasm API for this purpose.

Comment (hzxuzhonghu) [reply pilot/pkg/model/wasm.go:28]: i mean this flag should also be removed

Comment (howardjohn) [reply pilot/pkg/model/wasm.go:28]: oh thanks, missed that

**Review by ericvn — APPROVED** [review]

File: pilot/pkg/model/wasm.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/model/wasm.go:28]: done

**Review by zirain — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**